### ISSUE TITLE
Training the inverter

### ISSUE THREAD

**Issue by Jarvis658** [FusionBrainLab/StyleFeatureEditor#2]
How many steps does it take to train an inverter with full FFHQ data to achieve model convergence? Do you really need 300000 steps? Can you give a loss curve for reference? Thank you!

**Comment by retir** [comment]
Hi, thank you for your question! To acheive appeitiate inversion results inverter requers only about 15000 images (which means 1875 = 15000 / 8 training steps with batch size of 8) or even less. However, to reach metrics convergence, it indeed requires about 300k images (which means 37500 steps with batch size of 8). You can see it from our training graph: OX — number of seen images (30k-150k), OY — metric value.

![изображение_2024-06-24_222808577](https://github.com/AIRI-Institute/StyleFeatureEditor/assets/68122222/608c1e37-d45f-4da4-9895-db175b62f771)
![image](https://github.com/AIRI-Institute/StyleFeatureEditor/assets/68122222/ebecea79-cd68-483b-9980-42cb0046ac36)

**Comment by Jarvis658** [comment]
Thank you very much for your reply, which solved my doubts. But I also want to ask, are you using the full amount of FFHQ data? In other words, what is the amount of data in one of your epochs?

**Comment by retir** [comment]
We are not bound to epochs, we use infinite data loader (datasets/loaders.py), so we can validate\checpoint at any train step number we set. Inside the loader it just iterates over the full shuffled FFHQ dataset and when it comes to an end - just recreates the iterator. So if we call "epoch" the period between two iterator recreations, then we see every image of FFHQ (70k images) during one epoch.
